File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
       int p1 = 0;
       int p2 = numbers.length-1;
       int[] sol = new int[2];
       boolean c = false;
       while (!c){
        System.out.println("p1:"+p1);
        System.out.println("p2:"+p2);
            if (numbers[p1] + numbers[p2] == target){
                sol[0] = p1+1;
                sol[1] = p2+1;
                c = true;
                return sol;
            }
            if (numbers[p1] + numbers[p2] < target){
                p1++;
            }
            else if (numbers[p1] + numbers[p2] > target){
                p2--;
            }
       }
       return sol;
    }
}

File: Data Structures & Algorithms/eating-bananas/submission-1.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        p1 = 1
        p2 = max(piles)
        iters = 0
        best_targ = max(piles)
        while p1<=p2:
            iters+=1
            if iters==45:
                break
            total_h = 0
            target = (p2-p1)//2 + p1
            for i in piles:
                total_h += math.ceil(i/target)
            print(target)
            if target == 2:
                print(total_h)
            if total_h>h:
                p1 = target + 1
            elif total_h<=h:
                if target < best_targ: best_targ = target
                p2 = target - 1
        return best_targ